homeassistant/components/ping/binary_sensor.py

```diff
@@ -172,15 +171,12 @@ async def async_update(self) -> None:
         """Retrieve the latest details from the host."""
         _LOGGER.debug("ping address: %s", self._ip_address)
         try:
-            data = await self.hass.async_add_executor_job(
-                partial(
-                    icmp_ping,
-                    self._ip_address,
-                    count=self._count,
-                    timeout=ICMP_TIMEOUT,
-                    id=async_get_next_ping_id(self.hass),
```

Comment:
You no longer need to specify an identifier. icmplib now handles it automatically.